### DISCUSSION TITLE
How to install the new Google Analytics 4 code to your site?

### DISCUSSION THREAD

**Discussion by jarmos-san** [adityatelange/hugo-PaperMod#248] [General]
Google Analytics is now using the new `gtag.js` tracking code. But Hugo's internal GA template [doesn't work](https://github.com/gohugoio/hugo/issues/7954) with the new code yet. So, I just wanted to give a heads up to people who're not sure how to get it to work.

The code looks something like this - `G-XXX012XXYY`. And updating the `config.yaml` with that code instead of the older `UA-XXX012XXYY` wouldn't work. Fortunately & thanks for @adityatelange's forward thinking, you can easily integrate the new code with ease.

So, here's how you do it.

- Ensure you've updated your `config.yml` is updated with the new code. So, replace the older value of `GoogleAnalyticsID` with the new `G-XXX012XXYY`.
- Copy & paste the following `gtag.js` snippet into `extend_head.html`:

    ```javascript
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id={{ .Site.Params.GoogleAnalyticsID}}"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag() { dataLayer.push(arguments); }
        gtag('js', new Date());

        gtag('config', '{{ .Site.Params.GoogleAnalyticsID}}');
    </script>
    <!-- End of Global site tag (gtag.js) - Google Analytics -->
    ```
- Ensure your browser isn't blocking any JavaScript & tracking cookies while running the `hugo server` in the background. Check your site loads properly on `localhost:1313` & you should see something similar to the image below on your [Google Analytics dashboard](https://analytics.google.com/analytics/web).

    ![image](https://user-images.githubusercontent.com/31373860/108332311-a7bc6500-71f5-11eb-92bb-0abd85fed9c4.png)

**Comment by melissagwolf** [comment]
I just stumbled across this. I use a different template (`hugo-xmin`), but wanted to report that I was just able to add the new `GoogleAnalyticsID` with the new `G-XX` format to my `config.toml` file and begin successfully tracking visitors to my site without adding the `gtag.js` file. I'm not sure if this is template specific or if a change has been made so that the new GA4 tracking ID is now compatible with blogdown. Apologies if this does not apply.

**Reply by vdinolov** [reply]
@melissagwolf Could you share your solution? I am having the same issue and wondering if it is template specific...thanks!

**Reply by urwa** [reply]
After spending hours trying to add the js tags in the PaperMod submodule `extend_head.html`. I then read your comment and I only added the following in `config.yaml` and it worked just fine. I could see my click in Google Analytics when I loaded my site.
```
services:
  googleAnalytics:
    ID: G-XXXXXXXXX
```

**Reply by whiteboardmonk** [reply]
> After spending hours trying to add the js tags in the PaperMod submodule `extend_head.html`. I then read your comment and I only added the following in `config.yaml` and it worked just fine. I could see my click in Google Analytics when I loaded my site.
> 
> ```
> services:
>   googleAnalytics:
>     ID: G-XXXXXXXXX
> ```

This worked for me!

**Reply by satani99** [reply]
@whiteboardmonk i have tried everything i can think of but i somehow can't connect the google analytics to hugo papermod theme. 
can you please help me?

**Reply by whiteboardmonk** [reply]
> @whiteboardmonk i have tried everything i can think of but i somehow can't connect the google analytics to hugo papermod theme. can you please help me?

I've added a small block like this in my hugo.yml
![Screenshot 2024-11-19 at 9 44 59 AM](https://github.com/user-attachments/assets/89840eb7-5e55-4b3a-b462-5ed0371871b6)

You need any help in particular?

**Reply by satani99** [reply]
I have done the same but it's not working.
Are you hosting the site on GitHub pages? I'm hosting on that.

**Reply by duladissa** [reply]
@satani99 along with the below config in `hugo.yaml` you need to add the `google_analytics.html` as well

1. `hugo.yaml`

```yaml
services:
  googleAnalytics:
    id: G-XXXXXXXXXX
  instagram:
    disableInlineCSS: true
  twitter:
    disableInlineCSS: true
```

2. Create `layouts/_internal/google_analytics.html` and copy the content from here https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/google_analytics.html

3. Search in your project `google_analytics.html` and you should see  `themes/PaperMod/layouts/partials` as below screenshot

<img width="564" alt="Screenshot 2025-01-09 at 02 13 56" src="https://github.com/user-attachments/assets/b882fba7-3563-4de5-a6bb-555dc8c66cf3" />

**Comment by doneber** [comment]
I was trying and trying and it wasn't working until I noticed I was using Brave browser to test my site. Lol

**Comment by ytrepidorosonomous** [comment]
https://discourse.gohugo.io/t/replace-hugos-inbuilt-google-analytics-template-with-minimal-analytics-ga4/43704

**Comment by danieltak** [comment]
Should the documentation be updated to the [Hugo doc services](https://gohugo.io/methods/site/config/#services)?

Currently it is like this https://github.com/adityatelange/hugo-PaperMod/wiki/Installation:

```yaml
  analytics:
    google:
      SiteVerificationTag: "XYZabc"
    bing:
      SiteVerificationTag: "XYZabc"
    yandex:
      SiteVerificationTag: "XYZabc"
```

And seems to not work...

**Comment by duladissa** [comment]
Along with the below config in `hugo.yaml` you need to add the `google_analytics.html` as well

1. `hugo.yaml`

```yaml
services:
  googleAnalytics:
    id: G-XXXXXXXXXX
  instagram:
    disableInlineCSS: true
  twitter:
    disableInlineCSS: true
```

2. Create `layouts/_internal/google_analytics.html` and copy the content from here https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/google_analytics.html

3. Search in your project `google_analytics.html` and you should see  `themes/PaperMod/layouts/partials` as below screenshot

<img width="564" alt="Screenshot 2025-01-09 at 02 13 56" src="https://github.com/user-attachments/assets/b882fba7-3563-4de5-a6bb-555dc8c66cf3" />

**Reply by Kyu** [reply]
Thank you for including file paths
